test/component2_suggestors/defaultprops_initialstate_migrator_test.dart

```diff
@@ -0,0 +1,437 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/component2_suggestors/defaultprops_initialstate_migrator.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+
+main() {
+  migratorTests(
+      migratorName: 'GetDefaultPropsMigrator',
+      methodToMigrate: 'getDefaultProps');
+  migratorTests(
+      migratorName: 'GetInitialStateMigrator',
+      methodToMigrate: 'getInitialState');
+}
+
+void migratorTests({String migratorName, String methodToMigrate}) {
+  group('$migratorName', () {
+    componentDidUpdateTests(
+      methodToMigrate: methodToMigrate,
+    );
+  });
+
+  group('$migratorName with --no-partial-upgrades flag', () {
+    componentDidUpdateTests(
+      methodToMigrate: methodToMigrate,
+      allowPartialUpgrades: false,
+    );
+  });
+
+  group('$migratorName with --upgrade-abstract-components flag', () {
+    componentDidUpdateTests(
+      methodToMigrate: methodToMigrate,
+      shouldUpgradeAbstractComponents: true,
+    );
+  });
+
+  group(
+      '$migratorName with --no-partial-upgrades and --upgrade-abstract-components flag',
+      () {
+    componentDidUpdateTests(
+      methodToMigrate: methodToMigrate,
+      allowPartialUpgrades: false,
+      shouldUpgradeAbstractComponents: true,
+    );
+  });
+}
+
+void componentDidUpdateTests({
+  bool allowPartialUpgrades = true,
+  bool shouldUpgradeAbstractComponents = false,
+  methodToMigrate,
```

Comment:
This should be typed
```suggestion
  String methodToMigrate,
```